docs/production/upload-backends.md

```diff
@@ -55,6 +55,23 @@ uploading files, this process does not upload them to Amazon S3; see
 [migration instructions](#migrating-from-local-uploads-to-amazon-s3-backend)
 below for those steps.
 
+## S3 local caching
+
+For performance reasons, Zulip stores a cache of recently served user
+uploads on disk locally, even though the durable storage is kept in
+S3. There are a number of parameters which control the size and usage
+of this cache, which is maintained by nginx:
+
+- `s3_memory_cache_size` controls the in-memory size of the cache
+  _index_; the default is 1MB, which is enough to store about 8 thousand
+  entries.
+- `s3_disk_cache_size` controls the on-disk size of the cache
+  _contents_; the default is 100MB.
+- `s3_cache_inactive_time` controls the longest amount of time an
+  entry will be cached; the default is 1d. Since the contents of the
+  cache are immutable, this serves only as an additional limit on the
+  size of the contents on disk.
```

Comment:
Perhaps this discussion should have a bit of color on under what circumstances a sysadmin might want to change these parameters. I think probably the answer is that if you have a lot of disk space and send a lot of uploaded files, increasing `s3_disk_cache_size` to a few gigabytes is a good idea. 

I also wonder if we should make `s3_memory_cache_size` a function of the total system memory, like we do for things like `memcached` allocations; it's possible this number is small enough that it doesn't super make sense.

Reply:
Added a but of discussion about what the defaults are reasonable for, and when to increase them.